types/react-instantsearch-dom/server.d.ts

```diff
@@ -1,7 +1,7 @@
 import * as React from 'react';
 import { InstantSearchProps } from 'react-instantsearch-core';
 
-export function findResultsState(
-    App: React.ComponentType<Pick<InstantSearchProps, 'widgetsCollector'> & Record<string, any>>,
-    props: Pick<InstantSearchProps, 'searchClient' | 'indexName'> & Record<string, any>,
+export function findResultsState<TProps>(
+    App: React.ComponentType<TProps & InstantSearchProps>,
+    props: TProps & InstantSearchProps,
```

Comment:
```suggestion
    App: React.ComponentType<TProps & Pick<InstantSearchProps, 'widgetsCollector'>>,
    props: TProps & Pick<InstantSearchProps, 'searchClient' | 'indexName'>,
```

only these props are actually required, all others come from the props of the component :)

Reply:
IMO only certain parameters should be picked, others aren't used